org.scala-ide.sdt.core/src/org/scalaide/core/internal/async/AsyncDebuggerTab.scala

```diff
@@ -0,0 +1,123 @@
+package org.scalaide.core.internal.async
+
+import scala.util.Try
+
+import org.eclipse.core.runtime.CoreException
+import org.eclipse.debug.core.ILaunchConfiguration
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy
+import org.eclipse.debug.internal.ui.SWTFactory
+import org.eclipse.debug.ui.AbstractLaunchConfigurationTab
+import org.eclipse.jface.dialogs.Dialog
+import org.eclipse.pde.internal.ui.IHelpContextIds
+import org.eclipse.swt.SWT
+import org.eclipse.swt.events.SelectionAdapter
+import org.eclipse.swt.events.SelectionEvent
+import org.eclipse.swt.graphics.Image
+import org.eclipse.swt.layout.GridData
+import org.eclipse.swt.layout.GridLayout
+import org.eclipse.swt.widgets.Button
+import org.eclipse.swt.widgets.Composite
+import org.eclipse.ui.PlatformUI
+import org.scalaide.ui.ScalaImages
+
+import AsyncDebuggerConfiguration.LaunchWithAsyncDebugger
+
+private class GeneralSettingsBlock(val tab: AsyncDebuggerTab) {
+  import AsyncDebuggerConfiguration._
+
+  private var globalAsyncDebuggerSwitchButton: Button = _
+  val SettingsArea = "General Settings"
+  val AsyncDebuggerGlobalSwitch = "Switch On/Off Async Debugger functionality for this debug configuration"
+
+  def createControl(parent: Composite): Unit = {
+    val group = SWTFactory.createGroup(parent, SettingsArea, 1, 1, GridData.FILL_HORIZONTAL)
+    globalAsyncDebuggerSwitchButton = SWTFactory.createCheckButton(group, AsyncDebuggerGlobalSwitch, null, false, 1)
+    globalAsyncDebuggerSwitchButton.addSelectionListener(new SelectionAdapter() {
+      override def widgetSelected(e: SelectionEvent): Unit = {
+        tab.updateLaunchConfigurationDialog()
+      }
+    })
+  }
+
+  def setDefaults(configuration: ILaunchConfigurationWorkingCopy): Unit = {}
+
+  /**
+   * Initializes the state of UI components from the configuration attributes
+   * @param configuration the configuration to get attributes from
+   * @throws CoreException if an error occurs getting an attribute
+   */
+  def initializeFrom(configuration: ILaunchConfiguration): Unit =
+    globalAsyncDebuggerSwitchButton.setSelection(configuration.getAttribute(LaunchWithAsyncDebugger, false))
+
+  /**
+   * Sets attributes on the configuration based on the current state of the UI elements
+   * @param configuration configuration to modify
+   */
+  def performApply(configuration: ILaunchConfigurationWorkingCopy): Unit =
+    if (globalAsyncDebuggerSwitchButton.getSelection()) {
+      configuration.setAttribute(LaunchWithAsyncDebugger, true)
+    } else {
+      configuration.removeAttribute(LaunchWithAsyncDebugger)
+    }
+
+  /**
+   * @return a string error message or <code>null</code> if the block contents are valid
+   */
+  def validate: String = null
+}
+
+class AsyncDebuggerTab extends AbstractLaunchConfigurationTab {
```

Comment:
If we add a tab to the launch configuration, I would make it more general. A "Scala Debugger" tab that contains sections about different functionality is better for future additions.